### ISSUE TITLE
[Feature]request: Integration with caldav

### ISSUE THREAD

**Issue by andiyan1** [frappe/erpnext#2507]
Dear Team,

Hello, would there be any plan to integrate erpnext with caldav/google calendar and ical?
I know that some of us want the erpnext to be able to sync with google calendar and icalendar for apple.

This would enable erpnext to be able to seamlessly connect to the google for works platform and would enable us small business to fully leverage both platform of erpnext and google apps.

Thank you and best regards

Antonius

**Comment by RealDesignGraphics** [comment]
That would be great too ...  I'd be willing to sponsor at least part of the development

**Comment by shubhamgupta123** [comment]
Some work on this has already been done.
Repo: https://github.com/indictranstech/GCAL.git
Forum: https://discuss.erpnext.com/t/frappe-app-google-calendar-integration/6235
Hoping some developer would come forward and help with integrating this into the core.

**Comment by Tropicalrambler** [comment]
+1

**Comment by joolsr** [comment]
Yes +1 from me too

**Comment by woohoou** [comment]
+1

**Comment by mhbu50** [comment]
+1

**Comment by littlehera** [comment]
+1 outlook/office 365 calendar syncing would also be nice

**Comment by carlos22** [comment]
@littlehera sorry but most of the other systems pair with outlook and force you to pay the microsoft tax. CalDAV is an open standard which could be supported by microsoft but is not because they want you to use exchange server. It is possible to use caldav with outook extensions (one of them is free/libre and works well). CalDAV works with any device because most of the time you can get integrations apps/extensions (DAVDroid for Android, native in iOS/macOS and Thurnderbird, extension for Outook ...)

**Comment by dimitrisand** [comment]
Caldav  + cardav is a great combination and a great feature for an erp.

**Comment by lutealit** [comment]
+1

**Comment by carlos22** [comment]
Google Calendar seems to be available: https://frappe.io/blog/erpnext-features/erpnext-integrations
This is bad news as I see it - because it forces people who want to sync to use a proprietary closed US-Service which makes money with your data.

**Comment by joolsr** [comment]
This PR was put together by @chdecultot for a specific purpose of his from the community nor Frappe. From his history of writing a lot of code and providing it to the project I'm sure he only had good reasons behind this.
Still I agree a caldav connector would be also be very useful.

**Comment by chdecultot** [comment]
Hi,
Sorry I forgot to comment a few months ago.

Actually I do agree that a Caldav/Cardav solution would be better.
The Google Calendar Connector is a short/mid-term and pragmatic solution to be able to connect the calendar to en external target until someone funds the development of a better solution.

**Comment by Tropicalrambler** [comment]
I agree 100% with CarDAV and CalDAV functionality.  I found this neat little server, licensed under GPLv3 which seems like the perfect complement.  All that is necessary is to write the connectors to the Frappe based addresses and contacts list. I will be doing some tests to run it as a Custom App, and if development works, I will gladly propose it be merged into the core.

**Comment by parsimonyERP** [comment]
👍

**Comment by joolsr** [comment]
@Tropicalrambler I guess you would also need to write a connector for the Events doctype too?

**Comment by grauschnabel** [comment]
There are many ways which can be done. I would recommend just to implement it in the api, so users can use this as a link. I use nextcloud for that in the moment, which works pretty fine.  erpnext could do both, connect its calender and contacts to another server like nextcloud or radicale, where we sync it further. But I think better for the most people is to make it just available via the api.

**Comment by MarPiRK** [comment]
It would be just perfect if it was possible to sync ERPNext calendar and contacts to NextCloud.

**Comment by nandoschneider** [comment]
+1

**Comment by AminovE99** [comment]
+1

**Comment by N8jar** [comment]
+1

**Comment by s1risaac** [comment]
+1 for me a sync with kopano would be very interesting

**Comment by asoral** [comment]
+1 nextcloud maybe?

**Comment by geniusupgrader** [comment]
I just want to refer to #22919 , where the integration with nextcloud is discussed. There is already this app: https://github.com/doloresjuliana/pibiapp. Maybe its possible to add the functionality within this app to sync the contacts with nextcloud. Then one could sync the contacts further with the nextcloud calDAV and cardDAV servers...

**Comment by pibico** [comment]
Hello from Spain. I’m now working on a development on a Nextcloud Integration with Frappe/ERPNext (based on https://github.com/doloresjuliana/pibiapp which for the moment is only working on Frappe version-11 and not working on Frappe version-12 on).

I have developped (still in progress) a Frappe App to integrate CalDav NextCloud with Frappe. You can install from https://github.com/pibico/pibical (version-12 is more advanced than version-13 branch)

You should, first of all, on your user settings, inform about the credentials of your Nextcloud Caldav Server, User and Token. Once made, the Calendar Events show enhanced functionalities, such as Caldav Syncing (the event must be public for that).

The Sync with Caldav Server is made every minute. So, I have settled an event on iCalendar on my iPhone, and this has been synced with Nextcloud Calendar and then with Frappe, as you demanded. The other way around is working also, from Frappe is created a calendar Event on NextCloud Caldav Server and synced on iPhone.

Apart from that, the events have other functionalities such as minutes (taken the idea from the Frappe Meeting App). I’m working on future developments such as sending invitations with ics attachments from the event, or print format the minutes into pdf or word templates, but still will demand some working time.

Feel free to use and comment the bugs as this app is still under development.

**Comment by ankush** [comment]
Duplicate https://github.com/frappe/frappe/issues/339
